modules/abstract-utxo/src/abstractUtxoCoin.ts

```diff
@@ -1086,40 +1116,73 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
       isLastSignature = params.isLastSignature;
     }
 
-    if (_.isUndefined(userPrv) || !_.isString(userPrv)) {
-      if (!_.isUndefined(userPrv)) {
-        throw new Error(`prv must be a string, got type ${typeof userPrv}`);
+    const getSignerKeychain = (): utxolib.BIP32Interface => {
+      const userPrv = params.prv;
+      if (_.isUndefined(userPrv) || !_.isString(userPrv)) {
+        if (!_.isUndefined(userPrv)) {
+          throw new Error(`prv must be a string, got type ${typeof userPrv}`);
+        }
+        throw new Error('missing prv parameter to sign transaction');
       }
-      throw new Error('missing prv parameter to sign transaction');
-    }
+      const signerKeychain = bip32.fromBase58(userPrv, utxolib.networks.bitcoin);
+      if (signerKeychain.isNeutered()) {
+        throw new Error('expected user private key but received public key');
+      }
+      debug(`Here is the public key of the xprv you used to sign: ${signerKeychain.neutered().toBase58()}`);
+      return signerKeychain;
+    };
 
-    if (!params.pubs || params.pubs.length !== 3) {
-      throw new Error(`must provide xpub array`);
-    }
+    let signerKeychain: utxolib.BIP32Interface | undefined;
 
-    const signerKeychain = bip32.fromBase58(userPrv, utxolib.networks.bitcoin);
-    if (signerKeychain.isNeutered()) {
-      throw new Error('expected user private key but received public key');
+    if (tx instanceof bitgo.UtxoPsbt && isTransactionWithKeyPathSpendInput) {
+      switch (params.signingStep) {
+        case 'signerNonce':
+          signerKeychain = getSignerKeychain();
```

Comment:
We return anyway, no use setting the outer `let`

```suggestion
          const signerKeychain = getSignerKeychain();
```

Reply:
Another `getSignerKeychain` exists to justify it.